src/tokens/ERC1155/Packed/ERC1155PackedSale.sol

```diff
@@ -0,0 +1,295 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.17;
+
+import {IERC1155Sale} from "../IERC1155Sale.sol";
+import {ERC1155Metadata} from "@0xsequence/erc-1155/contracts/tokens/ERC1155/ERC1155Metadata.sol";
+import {ERC1155PackedSupply} from "./ERC1155PackedSupply.sol";
+import {ERC1155PackedBalance} from "@0xsequence/erc-1155/contracts/tokens/ERC1155PackedBalance/ERC1155PackedBalance.sol";
+import {ERC1155SaleErrors} from "../ERC1155SaleErrors.sol";
+import {ERC2981} from "@openzeppelin/contracts/token/common/ERC2981.sol";
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract ERC1155PackedSale is
```

Comment:
This duplicates a lot of code, I think we can do better if we make `ERC1155Sale` an abstract contract, then we can have two versions, one that inherits from the packed implementation and other that doesn't.

I'm not 100% sure that this will be possible, but we should explore it so we don't end up duplicating this much logic.